### PR TITLE
Directors Report API Approval 

### PR DESCRIPTION
**Pull request by faisal-amin82**
- Create directors report api approval controller and supporting classes
- Add unit tests for the controller and supporting classes

Resolves: SFA-2346

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/api/accounts/ResourceName.java [hunk 1]
```diff
@@ -24,7 +24,8 @@
     PROFIT_LOSS("profit-and-loss"),
     DIRECTORS_REPORT("directors-report"),
     DIRECTORS("directors"),
-    SECRETARY("secretary");
+    SECRETARY("secretary"),
+    DIRECTORS_REPORT_APPROVAL("directors-report-approval");
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/api/accounts/ResourceName.java]: Shall we use the existing `APPROVAL` enum?

File: src/main/java/uk/gov/companieshouse/api/accounts/controller/DirectorsApprovalController.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import uk.gov.companieshouse.api.accounts.utility.ErrorMapper;
+import uk.gov.companieshouse.api.accounts.utility.LoggingHelper;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping(value = "/transactions/{transactionId}/company-accounts/{companyAccountId}/small-full/directors-report/approval", produces = MediaType.APPLICATION_JSON_VALUE)
+public class DirectorsApprovalController {
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/api/accounts/controller/DirectorsApprovalController.java]: This class is missing update and delete?

File: src/main/java/uk/gov/companieshouse/api/accounts/service/impl/DirectorsApprovalService.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+            throw new DataException(e);
+        }
+
+        if (directorsApprovalEntity == null) {
+            return new ResponseObject<>(ResponseStatus.NOT_FOUND);
+        }
+
+        DirectorsApproval directorsApproval = transformer.transform(directorsApprovalEntity);
+        return new ResponseObject<>(ResponseStatus.FOUND, directorsApproval);
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/api/accounts/service/impl/DirectorsApprovalService.java]: You could use 
```
return new ResponseObject<>(ResponseStatus.FOUND, transformer.transform(directorsApprovalEntity));
```
here to remove the needless variable assignment

File: src/test/java/uk/gov/companieshouse/api/accounts/controller/DirectorsApprovalControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+
+        ResponseEntity responseEntity =
+                directorsApprovalController.update(directorsApproval,
+                        COMPANY_ACCOUNTS_ID, request);
+
+        assertNotNull(responseEntity);
+        assertEquals(HttpStatus.NOT_FOUND, responseEntity.getStatusCode());
+        assertNull(responseEntity.getBody());
+    }
+/*
```
Comment (BSaunders95) [inline src/test/java/uk/gov/companieshouse/api/accounts/controller/DirectorsApprovalControllerTest.java]: Should this be commented?

File: src/test/java/uk/gov/companieshouse/api/accounts/transformer/DirectorsApprovalTransformerTest.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+import java.util.HashMap;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class DirectorsApprovalTransformerTest {
```
Comment (BSaunders95) [inline src/test/java/uk/gov/companieshouse/api/accounts/transformer/DirectorsApprovalTransformerTest.java]: Should we assert the `date` field is mapped in this test class too?

File: src/test/java/uk/gov/companieshouse/api/accounts/controller/DirectorsApprovalControllerTest.java (hunk 4 above)
Comment (faisal-amin82) [reply src/test/java/uk/gov/companieshouse/api/accounts/controller/DirectorsApprovalControllerTest.java]: Removed the commented block as it is not required at the moment

File: src/main/java/uk/gov/companieshouse/api/accounts/service/impl/DirectorsApprovalService.java (hunk 3 above)
Comment (faisal-amin82) [reply src/main/java/uk/gov/companieshouse/api/accounts/service/impl/DirectorsApprovalService.java]: Removed the variable as suggested

File: src/test/java/uk/gov/companieshouse/api/accounts/transformer/DirectorsApprovalTransformerTest.java (hunk 5 above)
Comment (faisal-amin82) [reply src/test/java/uk/gov/companieshouse/api/accounts/transformer/DirectorsApprovalTransformerTest.java]: Added the code related to date assertion in directors approval transformer class

**Pull request merged**